.github/workflows/ci.yml

```diff
@@ -13,42 +13,67 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-18.04 ]
-        dmd: [ '2.092.*', '2.093.*' ]
+        dc: [ dmd-2.092.1, dmd-2.093.1 ]
         flavor: [ prod, devel ]
         include:
-          - { os: ubuntu-18.04, dmd: '2.092.*', dist: bionic, coverage: 0, closure_check: false }
-          - { os: ubuntu-18.04, dmd: '2.093.*', dist: bionic, coverage: 1, closure_check: true  }
+          - { dc: dmd-2.093.1, coverage: 1, closure_check: 1 }
 
     runs-on: ${{ matrix.os }}
     timeout-minutes: 30
     steps:
+    - name: "Ensure tools/ exists"
+      run: mkdir -p ${{ github.workspace }}/tools/
 
-    # Checkout this repository and its submodules
-    - uses: actions/checkout@v2
+    - name: 'Restore ebtree from cache'
+      id: cache-ebtree
+      uses: actions/cache@v1
       with:
-        submodules: true
+        path: ${{ github.workspace }}/tools/
+        key:  ebtree
 
-    - name: Setup environment
+    - name: 'Download ebtree from bintray'
+      if: steps.cache-ebtree.outputs.cache-hit != 'true'
       run: |
-        echo "$(git config -f .gitmodules submodule.beaver.path)/bin" >> $GITHUB_PATH
-        echo "COV=${{ matrix.coverage }}" >> $GITHUB_ENV
-        echo "DIST=${{ matrix.dist }}" >> $GITHUB_ENV
-        echo "DMD=${{ matrix.dmd }}" >> $GITHUB_ENV
-        echo "F=${{ matrix.flavor }}" >> $GITHUB_ENV
+        wget -O ${{ github.workspace }}/tools/libebtree.deb https://bintray.com/sociomantic-tsunami/dlang/download_file?file_path=libebtree6_6.0.s7-rc5-xenial_amd64.deb
+        wget -O ${{ github.workspace }}/tools/libebtree-dev.deb https://bintray.com/sociomantic-tsunami/dlang/download_file?file_path=libebtree6-dev_6.0.s7-rc5-xenial_amd64.deb
```

Comment:
What happens if you change these lines because the version is upgraded or something? Will this step be run even if there is a cache hit or do you have to somehow clear the cache manually?

Reply:
If there's a version change, I'll probably update the `key` to include the distribution, so there will be no cache hit.
We can also clear the cache manually if we wish, but the `key` approach works best IMO.
Example usage (although it's a bit useless since it only runs on `windows`): https://github.com/dlang/druntime/blob/306bd965ea9d83bad7e5444ff9d5e1af1a6d934a/.github/workflows/cxx.yml#L220-L227